test_conformance/api/test_queue_array_properties.cpp

```diff
@@ -0,0 +1,273 @@
+﻿//
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+
+#include "testBase.h"
+#include "harness/typeWrappers.h"
+#include <sstream>
+#include <string>
+#include <vector>
+#include <algorithm>
+
+struct test_queue_array_properties_data
+{
+    std::vector<cl_queue_properties> properties;
+    std::string description;
+};
+
+int verify_if_properties_supported(
+    cl_device_id deviceID, cl_command_queue_properties requested_bitfield,
+    cl_uint requested_size)
+{
+    int error = CL_SUCCESS;
+    cl_command_queue_properties on_host_properties = 0;
+    cl_device_device_enqueue_capabilities device_enqueue_caps;
+    std::vector<cl_command_queue_properties> all_properties = {
+        CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, CL_QUEUE_PROFILING_ENABLE,
+        CL_QUEUE_ON_DEVICE, CL_QUEUE_ON_DEVICE_DEFAULT
+    };
+
+    error =
+        clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_ON_HOST_PROPERTIES,
+                        sizeof(on_host_properties), &on_host_properties, NULL);
+    test_error(error,
+               "clGetDeviceInfo for CL_DEVICE_QUEUE_ON_HOST_PROPERTIES failed");
+
+    error = clGetDeviceInfo(deviceID, CL_DEVICE_DEVICE_ENQUEUE_CAPABILITIES,
+                            sizeof(device_enqueue_caps), &device_enqueue_caps,
+                            NULL);
+    test_error(
+        error,
+        "clGetDeviceInfo for CL_DEVICE_DEVICE_ENQUEUE_CAPABILITIES failed");
+    if (!(device_enqueue_caps & CL_DEVICE_QUEUE_SUPPORTED)
+        && requested_size > 0)
```

Comment:
Minor: This test has chosen a small enough value for the queue size that it _probably_ will be supported for all devices, but to be sure it could also skip itself if the `requested_size` is greater than `CL_DEVICE_QUEUE_ON_DEVICE_MAX_SIZE`.